Clinical-Genomics/scout
# Reassign MME cases to another user when a Scout user is removed

Fix #2979

**How to test**:
1. In a terminal, start a MatchMaker Exchange server with demo data:
```
git clone https://github.com/Clinical-Genomics/patientMatcher.git
cd patientMatcher
docker-compose up
```
2. In a local instance of Scout, demo config file (config.py), uncomment these lines: 
```
MME_ACCEPTS = "application/vnd.ga4gh.matchmaker.v1.0+json"
MME_URL = "http://localhost:8000"
MME_TOKEN = "DEMO"
```
3. Modify demo user and grant him permission to share cases to MatchMaker:
```
scout --demo update user -u clark.kent@mail.com -r mme_submitter
```
4. Launch the web app of Scout, pin a variant and share case to the MatchMaker.
![image](https://user-images.githubusercontent.com/28093618/142411491-7a80a294-e41f-4ca1-828c-07ac9201659f.png)

5. Check in MongoDB compass, where a submission object dictionary should have been created for the demo case. Note that the submitter **Clark Kent is present twice in this object: both in patient.contact as an object and its mail under subm_user**

![image](https://user-images.githubusercontent.com/28093618/142411642-b977c1d6-868d-47a9-96b7-82970eebbfe1.png)


6. Stop the Scout server and try to delete the demo user with this command:
```
scout --demo delete user -m clark.kent@mail.com
```
7. It should complain and ask you if you want to reassign the MME cases:

![image](https://user-images.githubusercontent.com/28093618/142412289-3461ee24-4efd-413b-aa94-e14f8aaf8a1b.png)

8. If you type N, then the command should exit.
9. If you type Y, the it will ask you the email for the new MME contact, a user already present in Scout.
10. Create a new user and repeat, try to assign the MME case to this user. It should fail if the new user doesn't have the "mme_submitter" role
11. After updating the user and granting them the `mme_submitter` (as in 3), the command should be successful and:
- The case should be reassigned to the new user (check the case document in mongo compass)
- Demo user Clark Kent should have been removed.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
